vertx-mssql-client/src/main/java/io/vertx/mssqlclient/impl/codec/DataType.java

```diff
@@ -152,7 +155,27 @@ public Object decodeValue(ByteBuf byteBuf, Metadata metadata) {
       return byteBuf.readFloatLE();
     }
   },
-  MONEY(0x3C),
+  MONEY(0x3C) {
+    @Override
+    public Metadata decodeMetadata(ByteBuf byteBuf) {
+      return null;
+    }
+
+    @Override
+    public Object decodeValue(ByteBuf byteBuf, Metadata metadata) {
+      ByteBuffer bb = ByteBuffer.allocate(8);
+      bb.order(ByteOrder.BIG_ENDIAN);
+      bb.putInt(byteBuf.readIntLE());
+      bb.putInt(byteBuf.readIntLE());
+
+      return new BigDecimal(new BigInteger(bb.array())).divide(new BigDecimal("10000"), 4, RoundingMode.UP);
```

Comment:
This could be simplified to avoid the allocation:

```suggestion
      long highBits = (long) byteBuf.readIntLE() << 32;
      long lowBits = byteBuf.readIntLE() & 0xFFFFFFFFL;
      BigInteger bigInteger = BigInteger.valueOf(highBits | lowBits);
      return new BigDecimal(bigInteger).divide(new BigDecimal("10000"), 4, RoundingMode.UP);
```

Reply:
Looks super!